x/tokenfactory/keeper/before_send_test.go

```diff
@@ -115,3 +241,99 @@ func (suite *KeeperTestSuite) TestNonWhitelistedHooksNotCalled() {
+	for _, tc := range []struct {
+		name            string
+		wasmFile        string
+		tokenToSend     sdk.Coins
+		useFactoryDenom bool
+		blockBeforeSend bool
+		expectedError   bool
+	}{
+		{
+			name:            "sending tokenfactory denom from module to module with infinite contract should panic",
+			wasmFile:        "./testdata/infinite_track_beforesend.wasm",
+			useFactoryDenom: true,
+		},
+		{
+			name:            "sending tokenfactory denom from account to account with infinite contract should panic",
+			wasmFile:        "./testdata/infinite_track_beforesend.wasm",
+			useFactoryDenom: true,
+			blockBeforeSend: true,
+		},
+		{
+			name:            "sending non-tokenfactory denom from module to module with infinite contract should not panic",
+			wasmFile:        "./testdata/infinite_track_beforesend.wasm",
+			tokenToSend:     sdk.NewCoins(sdk.NewInt64Coin("foo", 1000000)),
+			useFactoryDenom: false,
+		},
+		{
+			name:            "Try using no 100",
+			wasmFile:        "./testdata/no100.wasm",
+			useFactoryDenom: true,
+		},
+	} {
+		suite.Run(fmt.Sprintf("Case %suite", tc.name), func() {
+			// setup test
+			suite.Setup()
+
+			// load wasm file
+			wasmCode, err := os.ReadFile(tc.wasmFile)
+			suite.Require().NoError(err)
+
+			// instantiate wasm code
+			codeID, _, err := suite.contractKeeper.Create(suite.ChainA.GetContext(), suite.ChainA.SenderAccounts[0].SenderAccount.GetAddress(), wasmCode, nil)
+			suite.Require().NoError(err, "test: %v", tc.name)
+			cosmwasmAddress, _, err := suite.contractKeeper.Instantiate(suite.ChainA.GetContext(), codeID, suite.TestAccs[0], suite.ChainA.SenderAccounts[0].SenderAccount.GetAddress(), []byte("{}"), "", sdk.NewCoins())
+			suite.Require().NoError(err, "test: %v", tc.name)
+
+			params := types.DefaultParams()
+			params.WhitelistedHooks = []*types.WhitelistedHook{{DenomCreator: suite.ChainA.SenderAccounts[0].SenderAccount.GetAddress().String(), CodeID: 1}}
+			err = suite.GetNeutronZoneApp(suite.ChainA).TokenFactoryKeeper.SetParams(suite.ChainA.GetContext(), params)
+			suite.Require().NoError(err)
+
+			// create new denom
+			res, err := suite.msgServer.CreateDenom(suite.ChainA.GetContext(), types.NewMsgCreateDenom(suite.ChainA.SenderAccounts[0].SenderAccount.GetAddress().String(), "bitcoin"))
+			suite.Require().NoError(err, "test: %v", tc.name)
+			factoryDenom := res.GetNewTokenDenom()
+
+			var tokenToSend sdk.Coins
+			if tc.useFactoryDenom {
+				tokenToSend = sdk.NewCoins(sdk.NewInt64Coin(factoryDenom, 100))
+			} else {
+				tokenToSend = tc.tokenToSend
+			}
+
+			// send the mint module tokenToSend
+			if tc.blockBeforeSend {
+				suite.FundAcc(suite.ChainA.SenderAccounts[0].SenderAccount.GetAddress(), tokenToSend)
+			} else {
+				suite.FundModuleAcc(icqtypes.ModuleName, tokenToSend)
+			}
+
+			// set beforesend hook to the new denom
+			// we register infinite loop contract here to test if we are gas metering properly
```

Comment:
As far as i can see, no100 in the list of testcases and it's not `infinite loop`. Comment misleading?